scout/server/blueprints/alignviewers/views.py

```diff
@@ -50,117 +51,39 @@ def unindexed_remote_static():
-
-    display_obj = {}
-
-    display_obj["reference_track"] = controllers.reference_track(chromosome_build, chrom)
-    display_obj["genes_track"] = controllers.genes_track(chromosome_build, chrom)
-    display_obj["clinvar_snvs"] = controllers.clinvar_track(chromosome_build, chrom)
-    display_obj["clinvar_cnvs"] = controllers.clinvar_cnvs_track(chromosome_build, chrom)
-
-    # Init upcoming igv-tracks
-    sample_tracks = []
-    upd_regions_bed_tracks = []
-    upd_sites_bed_tracks = []
-
-    counter = 0
-    for sample in samples:
-        # some samples might not have an associated bam file, take care if this
-        if len(bam_files) > counter and bam_files[counter]:
-            sample_tracks.append(
-                {
-                    "name": sample,
-                    "url": bam_files[counter],
-                    "format": bam_files[counter].split(".")[-1],  # "bam" or "cram"
-                    "indexURL": bai_files[counter],
-                    "height": 700,
-                }
-            )
-        counter += 1
-
-    display_obj["sample_tracks"] = sample_tracks
-
-    if rhocall_wig_files:
-        rhocall_wig_tracks = make_igv_tracks("Rhocall Zygosity", rhocall_wig_files)
-        display_obj["rhocall_wig_tracks"] = rhocall_wig_tracks
-    if rhocall_bed_files:
-        rhocall_bed_tracks = make_igv_tracks("Rhocall Regions", rhocall_bed_files)
-        display_obj["rhocall_bed_tracks"] = rhocall_bed_tracks
-    if tiddit_coverage_files:
-        tiddit_wig_tracks = make_igv_tracks("TIDDIT Coverage", tiddit_coverage_files)
-        display_obj["tiddit_wig_tracks"] = tiddit_wig_tracks
-    if updregion_files:
-        updregion_tracks = make_igv_tracks("UPD region", updregion_files)
-        display_obj["updregion_tracks"] = updregion_tracks
-    if updsites_files:
-        updsites_tracks = make_igv_tracks("UPD sites", updsites_files)
-        display_obj["updsites_tracks"] = updsites_tracks
-
-    if request.form.get("center_guide"):
-        display_obj["display_center_guide"] = True
-    else:
-        display_obj["display_center_guide"] = False
+    display_obj = {}  # Initialize the dictionary containing all tracks info
 
-    return render_template("alignviewers/igv_viewer.html", locus=locus, **display_obj)
+    # Set up IGV tracks that are common for all cases:
+    display_obj["reference_track"] = HUMAN_REFERENCE[
+        chromosome_build
+    ]  # Human reference is always present
+    # General tracks (Genes, Clinvar and ClinVar SNVs are shown according to user preferences)
+    controllers.set_common_tracks(display_obj, chromosome_build, request.form)
 
+    # Set up bam/cram alignments for case samples:
+    controllers.set_sample_tracks(display_obj, request.form)
 
-def make_igv_tracks(name, file_list):
-    """ Return a dict according to IGV track format. """
+    # Set up case-specific tracks (might be present according to the pipeline)
+    if request.form.get("bam"):
```

Comment:
Hm, will that really always be set? E.g. UPDsites or TIDDIT_coverage still left on file system with an old archived bam?

Reply:
Isn't this a problem we also have now in current master then? I don't think we ever reach this function (alignviewers) if there is no bam or MT_bam assigned for the case individual..

Reply:
😆 True enough! Makes one one wonder why the check is there?

Reply:
But you raised an important issue here, we should fix it for when there is no alignment file but other tracks are still available

Reply:
But in another PR perhaps?

Reply:
But this if might be perhaps removed in this PR, I'll fix it

Reply:
I think that code was like this originally to prepare those tracks only if the chromosome is not MT, you know? That's why I left the condition, now I remember. Is this the case? -- > do you want these tracks loaded also when igv is showing MT alignments?

Reply:
A yes, right. I remember I used that as a workaround sometimes when looking at the other tracks locally, but not having a full bam, only an MT one. No, agreed, that is good. But perhaps make that clear so we don't loose it at a later stage!

Reply:
I'll change the condition to check if chrom != MT instead, so it's more clear

Reply:
Done